src/gg.regression.unity.bots/Runtime/Scripts/Validation/RGValidationScript.cs

```diff
@@ -0,0 +1,251 @@
+                    currentValidator = validator;
+                    validator.Method.Invoke(this, null);
+                } 
+                catch (Exception e)
+                {
+                    Debug.LogError($"Error running validation method {validator.Method.Name}: {e.Message}");
+                    validator.ThrownException = e;
+                }
+                
+                // After the method is called, we can figure out what to do with the validators that should immediately
+                // fail or pass
+                if (currentValidator.Mode == ValidationMode.NEVER_TRUE && currentStatus == SegmentValidationStatus.PASSED)
+                {
+                    // Debug.LogError($"Validation method {validator.Method.Name} should never pass, but it did.");
+                    currentValidator.Status = SegmentValidationStatus.FAILED;
+                    OnValidationsUpdated?.Invoke();
+                }
+                else if (currentValidator.Mode == ValidationMode.ALWAYS_TRUE && currentStatus is SegmentValidationStatus.FAILED or SegmentValidationStatus.UNKNOWN)
+                {
+                    // Debug.LogError($"Validation method {validator.Method.Name} should always pass, but it did not.");
+                    currentValidator.Status = SegmentValidationStatus.FAILED;
+                    OnValidationsUpdated?.Invoke();
+                } 
+                else if (currentValidator.Mode == ValidationMode.EVENTUALLY_TRUE && currentStatus == SegmentValidationStatus.PASSED)
+                {
+                    // Debug.LogError($"Validation method {validator.Method.Name} finally passed.");
+                    currentValidator.Status = SegmentValidationStatus.PASSED;
+                    OnValidationsUpdated?.Invoke();
+                }
+                else if (currentValidator.Mode == ValidationMode.ONCE_TRUE_ALWAYS_TRUE &&
+                         currentValidator.Status == SegmentValidationStatus.PASSED &&
+                         currentStatus != SegmentValidationStatus.PASSED)
+                {
+                    // Debug.LogError($"Validation method {validator.Method.Name} passed before but is no longer passing.");
+                    currentValidator.Status = SegmentValidationStatus.FAILED;
+                    OnValidationsUpdated?.Invoke();
+                }
+                else if (currentValidator.Mode == ValidationMode.ONCE_TRUE_ALWAYS_TRUE &&
+                         currentValidator.Status == SegmentValidationStatus.UNKNOWN &&
+                         currentStatus == SegmentValidationStatus.PASSED)
+                {
+                    // Debug.LogError($"Validation method {validator.Method.Name} is now passing.");
+                    currentValidator.Status = SegmentValidationStatus.PASSED;
+                    OnValidationsUpdated?.Invoke();
+                } 
+                else if (currentValidator.Mode == ValidationMode.ONCE_TRUE_ALWAYS_TRUE &&
```

Comment:
Courtesy of O1:

Here, if the validator has never passed before (it is UNKNOWN) and the current check fails, you immediately mark it as FAILED. This could be problematic depending on the intended semantics:

The documentation states: "Once there is a frame that the given condition turns true, it must always be marked as true." It does not explicitly say that if the condition fails before ever becoming true, it should never have a chance to pass later.

By failing immediately on a single failed invocation before a pass, you deny the validator any future opportunity to eventually pass. This might not be what you want logically.

Reply:
You are correct. I have removed this check!